### ISSUE TITLE
python3Packages.setuptools: bootstrapping changed

### ISSUE THREAD

**Issue by FRidh** [NixOS/nixpkgs#117014]
**Describe the bug**
As of 53.0.0 the bootstrap.py has been removed. Building entirely from source (non-wheels) is going to be challenging now.
- https://github.com/pypa/setuptools/blob/main/CHANGES.rst#v5300
- https://github.com/pypa/setuptools/issues/1527
- https://github.com/pypa/setuptools/pull/2543

**Notify maintainers**
cc @jonringer @dotlambda

**Comment by FRidh** [comment]
We can keep the old bootstrap script https://github.com/pypa/setuptools/pull/2543#issuecomment-803276594

**Comment by FRidh** [comment]
cc @adisbladis @DavHau , this will be relevant for you as well with newer setuptools.

**Comment by adisbladis** [comment]
@FRidh Thanks for the heads up! I think `poetry2nix` is unaffected since we use `setuptools` from nixpkgs.

**Comment by jonringer** [comment]
I say, we just vendor it, and then use it before the `buildPhase`

**Comment by jonringer** [comment]
I had issues with getting `boostrapped-pip` to build when trying to just vendor the bootstrap script. 

```
++ /nix/store/555digis0k4642hmvd0pwzvc2aw5gn09-python3-3.8.8/bin/python3.8 -m pip install --no-build-isolation --no-index --prefix=/nix/store/q86whd0vvfyblyq53az37233vaklqw7f-python3.8-bootstrapped-pip-21.0.1 --ignore-installed --no-dependencies --no-cache .
Processing /build/setuptools
    Preparing wheel metadata ... error
    ERROR: Command errored out with exit status 1:
     command: /nix/store/555digis0k4642hmvd0pwzvc2aw5gn09-python3-3.8.8/bin/python3.8 /build/pip/src/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /build/tmps7c2loln
         cwd: /build/pip-req-build-yku5fwwx
    Complete output (24 lines):
    running dist_info
    creating /build/pip-modern-metadata-js1r307r/setuptools.egg-info
    writing /build/pip-modern-metadata-js1r307r/setuptools.egg-info/PKG-INFO
    writing dependency_links to /build/pip-modern-metadata-js1r307r/setuptools.egg-info/dependency_links.txt
    writing entry points to /build/pip-modern-metadata-js1r307r/setuptools.egg-info/entry_points.txt
    writing requirements to /build/pip-modern-metadata-js1r307r/setuptools.egg-info/requires.txt
    writing top-level names to /build/pip-modern-metadata-js1r307r/setuptools.egg-info/top_level.txt
    writing manifest file '/build/pip-modern-metadata-js1r307r/setuptools.egg-info/SOURCES.txt'
    package init file 'setuptools/tests/indexes/__init__.py' not found (or not a regular file)
    package init file 'setuptools/tests/indexes/test_links_priority/__init__.py' not found (or not a regular file)
    package init file 'setuptools/tests/indexes/test_links_priority/simple/__init__.py' not found (or not a regular file)
    package init file 'setuptools/tests/indexes/test_links_priority/simple/foobar/__init__.py' not found (or not a regular file)
    package init file 'pkg_resources/tests/data/__init__.py' not found (or not a regular file)
    package init file 'pkg_resources/tests/data/my-test-package-source/__init__.py' not found (or not a regular file)
    package init file 'pkg_resources/tests/data/my-test-package_zipped-egg/__init__.py' not found (or not a regular file)
    package init file 'pkg_resources/tests/data/my-test-package_unpacked-egg/__init__.py' not found (or not a regular file)
    package init file 'pkg_resources/tests/data/my-test-package-zip/__init__.py' not found (or not a regular file)
    reading manifest file '/build/pip-modern-metadata-js1r307r/setuptools.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    warning: no files found matching '*.py' under directory 'tests'
    warning: no files found matching 'Makefile' under directory 'docs'
    writing manifest file '/build/pip-modern-metadata-js1r307r/setuptools.egg-info/SOURCES.txt'
    creating '/build/pip-modern-metadata-js1r307r/setuptools.dist-info'
    error: invalid command 'bdist_wheel'
```

I'm assuming this is because we put pip, wheel, and setuptools in separate directories, and then the wheel commands will never be imported, as setuptools will be traversed first. But I'm not sure how it was working previously.

I guess I could have tried just moved the 3 sources into a single directory, and see if that could have worked, but I already deleted all the work because I was getting frustrated :)

**Comment by FRidh** [comment]
> But I'm not sure how it was working previously.

setuptools supports pep517 again, so there is a pyproject.toml that takes precedence. Removing it gives the old situation.

**Comment by FRidh** [comment]
Fixed on staging with 252ae444165fe4be8c8b3e3fc42a1c4eb857d609 and 983c453eec9d45cebcf898cf33d87c6fc78e05ad.
